model_params:
  weights_path: "./traversability_model_trace_distill128_cfs_rgbonly.pt"
  static_output_dim: 32
  modality: "monocular"
map_params:
  resolution: 0.1
  height: 25.6
  width: 25.6
  origin_x: 12.8
  origin_y: 12.8
  # map_to_world: [0.0, 5.0, -2.0]
  map_to_base_link: [0.0, 0.0, 0.0] # Note we denote world=base_link
planner_params:
  max_v: 1.0
  # max_w: 1.57079
  max_w: 0.3
  # max_w: 0.0 
  max_dv: 0.2
  # max_dw: 0.78539
  max_dw: 0.15
  partitions: 11
  dt: 1.0
  max_iters: 10000
  # Set max time to 0.0 to not use 
  max_time: 0.0
  goal_tolerance: 0.5
  cost_weight: 1.5
input_topics: # Uses these to determine
  left_image: "/stereo/left/image_raw/compressed"
  left_camera_info: "/stereo/left/camera_info"
  right_image: ""
  right_camera_info: ""
  pointcloud: ""
output_topics:
  depth: "/stereo/left/image_raw/depth"
  traversability: "/creste/bev/traversability"
  semantic_elevation: "/creste/3d/semantic_elevation"
viz_params:
  history_window: 3
  viz_3d: False

pt2pix: # LiDAR to Rectified Camera Pixels (Full resolution)
  rows: 3
  cols: 4
  # data: [ # 2024-11-17
  #   295.64482, -380.52208, -44.44967, 92.68059,
  #   195.38800, 7.27873, -411.18572, 62.37294,
  #   0.99153, 0.00623, -0.12966, 0.10123
  # ]
  # data: [ #2025-01-10
  #   305.22870, -371.22992, -32.38245, 87.04788,
  #   208.37674, 4.84756, -397.81979, 70.21586,
  #   0.99492, 0.01250, -0.09979, 0.09980
  # ]
  # data: [ # 2025-01-22
  #   309.143910, -387.72699, -19.77981, 81.43987,
  #   231.54858, 7.85995, -397.57267, 71.48898,
  #   0.99921, 0.00125, -0.03950, 0.08004
  # ]
  # data: [ # 2025-01-23 morning
  #   308.90741, -369.29905, -15.67597, 68.23324,
  #   230.65223, 7.48190, -385.80649, 57.79381, 0.99887,
  #   0.02050, -0.04286, 0.065264
  # ]
  # data: [ # 2025-01-23 afternoon (hot glue)
  #   308.60880, -371.43606, -18.16256, 73.28075,
  #   232.43691, 7.92877, -388.54405, 61.57596,
  #   0.99869, 0.01813, -0.04782, 0.07770
  # ]
  # # data: [ # 2025-01-27 long horizon 
  #   309.66738316382714, -368.6516473532315, -18.9832257034209, 80.10430530064465,
  #   225.31604282915657, 7.018504524287742, -388.9854080497502, 73.79079945723205,
  #   0.9981371920857036, 0.022694649578873943, -0.05663125158213658, 0.10704920229509099
  # ]
  # data: [ # 2025-02-09 long horizon
  #   312.2734436351036, -371.6243150391048, -28.048451279228175, 63.526840559151694,
  #   219.15571073622982, 10.872136676481922, -394.9965066662114, 64.07488761843501,
  #   0.9971948273231721, 0.021436158592018077, -0.07171448573842666, 0.06059922069105455
  # ]
  data: [ # 2025-02-24 stereo (hot glue)
    311.097130696863, -368.86113960450564, -20.472152923184833, 73.32105286297515,
    222.3181943734818, 5.815265363826277, -390.7515741323514, 70.05054410720905, 
    0.9976823262605736, 0.021959678709886483, -0.06440301528848538, 0.08707664590138037
  ]
pix2pt: # Rectified Camera Pixels to LiDAR (Downsampled)
  rows: 4
  cols: 4
  # data: [ # 2024-11-17
  #   0.0000443, -0.0013571,  1.0720886, -0.0883973,
  #   -0.0104583,  0.0001569,  0.7718496,  0.1617347,
  #   -0.0001641, -0.0103701,  0.5231002,  0.1125486,
  #   0.,         0.,         0.,         1.
  # ]
  # data: [ # 2025-01-10
  #   0.000128,     -0.001065,      1.051069,     -0.088983,
  #   -0.010665,      0.000050,      0.815308,      0.149835,
  #   -0.000063,     -0.010612,      0.560481,      0.131719,
  #   -0.000000,      0.000000,      0.000000,      1.000000
  # ]
  # data: [ # 2025-01-22
  #   0.000005,  -0.0004073,  1.0239925, -0.0747868,
  #   -0.0103023,  0.0002004,  0.7852374,  0.1433194,
  #   -0.0002008, -0.0102943,  0.6119031,  0.1390907,
  #   0.0,         0.0,         0.0,         1.0       
  # ]
  # data: [ # 2025-01-23 morning
  #   0.0002152, -0.000458,   1.0109346, -0.0630307,
  #   -0.010648,   0.0000685,  0.8192868,  0.1271771,
  #   -0.0000778, -0.0106404,  0.6202699,  0.1145838,
  #   0.,         0.,         0.,         1.
  # ]
  # data: [ # 2025-01-23 afternoon (hot glue)
  #   0.0001876, -0.0005092,  1.0164443, -0.0745763,
  #   -0.010608,   0.0000951,  0.8139705,  0.1296312,
  #   -0.0001042, -0.0105975,  0.6246729,  0.1165106,
  #   0.,         0.,         0.,         1.
  # ]
  # data: [ # 2025-01-27 long horizon 
  #   0.000239,  -0.0006042,  1.0174218, -0.1025557,
  #   -0.0106468,  0.00004,    0.8235223,  0.1243182,
  #   -0.0000536, -0.0106324,  0.6041906,  0.1325394,
  #   0.,         0.,         0.,         1. 
  # # ]
  # data: [ # 2025-02-09 long horizon
  #   0.0002149, -0.0007617,  1.0278419, -0.0534973,
  #   -0.01057,    0.0001558,  0.8189456,  0.1157467,
  #   -0.0001717, -0.010545,   0.5928182,  0.1357203,
  #   0.,         0.,         0.,         1.
  # ]
  data: [ # 2025-02-24 stereo (hot glue)
    0.0002327, -0.0006862,  1.0224161, -0.0812753,
    -0.0106465,  0.000011,   0.8293346,  0.1227447,
    -0.0000261, -0.010627,   0.5940462,  0.1348564,
    0.,         0.,         0.,         1.
  ]